partiql-conformance-tests/tests/test_value.rs

```diff
@@ -0,0 +1,231 @@
+        IonType::String => Value::String(Box::new(reader.read_string().unwrap())),
+        IonType::Clob => todo!("clob"),
+        IonType::Blob => Value::Blob(Box::new(reader.read_blob().unwrap())),
+        IonType::List => {
+            if has_annotation(reader, "$bag") {
+                Bag::from(parse_test_value_sequence(reader)).into()
+            } else {
+                List::from(parse_test_value_sequence(reader)).into()
+            }
+        }
+        IonType::SExpression => todo!("sexp"),
+        IonType::Struct => parse_test_value_tuple(reader).into(),
+    }
+}
+
+fn parse_test_value_tuple(reader: &mut Reader) -> Tuple {
+    let mut tuple = Tuple::new();
+    reader.step_in().expect("step into struct");
+    while let item = reader.next().expect("struct value") {
+        let (key, value) = match item {
+            StreamItem::Value(typ) => (
+                reader.field_name().expect("field name"),
+                parse_test_value(reader, typ),
+            ),
+            StreamItem::Null(_) => (
+                reader.field_name().expect("field name"),
+                parse_null(&reader),
+            ),
+            StreamItem::Nothing => break,
+        };
+        tuple.insert(key.text().unwrap(), value);
+    }
+    reader.step_out().expect("step out of struct");
+    tuple
+}
+
+fn parse_test_value_sequence(reader: &mut Reader) -> Vec<Value> {
+    reader.step_in().expect("step into sequence");
+    let mut values = vec![];
+    loop {
+        let item = reader.next().expect("test value");
+        let val = match item {
+            StreamItem::Value(typ) => parse_test_value(reader, typ),
+            StreamItem::Null(_) => parse_null(&reader),
+            StreamItem::Nothing => break,
+        };
+        values.push(val);
+    }
+    reader.step_out().expect("step out of sequence");
+    values
+}
+
+#[cfg(test)]
+#[cfg(not(feature = "conformance_test"))]
+mod tests {
+    use super::parse_test_value_str;
+    use ion_rs::Decimal;
+    use partiql_value::{partiql_bag, partiql_list, partiql_tuple, Bag, List, Tuple, Value};
+
+    #[track_caller]
+    fn parse(test: &str, expected: Value) {
+        let val = parse_test_value_str(test);
+        assert_eq!(val, expected);
+    }
+
+    #[test]
+    fn simple() {
+        parse("null", Value::Null);
+        parse("$missing::null", Value::Missing);
+        parse("9", Value::Integer(9));
+        parse("true", Value::Boolean(true));
+        parse("false", Value::Boolean(false));
+        parse("\"str\"", Value::String(Box::new("str".into())));
+    }
+
+    #[test]
+    fn bag() {
+        let test = "$bag::[
+            {
+                f: 1,
+                d: 2e0,
+                s: 1
+            }
+        ]";
+        let expected = Value::from(partiql_bag![partiql_tuple![
+            ("f", 1),
+            ("d", Value::Real(2.0.into())),
+            ("s", 1)
+        ]]);
```

Comment:
No assertion here currently. Should we add a call to `parse` with `test` and `expected`?